pkg/cleaner/jobcontroller.go

```diff
@@ -75,10 +74,20 @@ func NewCleanupJob(bd *v1alpha1.BlockDevice, volMode VolumeMode, tolerations []v
 	mountName := "vol-mount"
 
 	if volMode == VolumeModeBlock {
+		jobContainer.Command = []string{"/bin/sh", "-c"}
+
+		pathRegex := bd.Spec.Path
+		// if deviceType is a disk, there can be partitions on the disk. In that case
+		// partitions need to be wiped before wiping main disk
+		if bd.Spec.Details.DeviceType == "disk" {
+			pathRegex = bd.Spec.Path + "*"
```

Comment:
can we traverse the children and wipe the device path here?

Reply:
You mean, for `sdb`, traversing through `sdb1`, `sdb2`... etc and wiping them right?

Reply:
Yes, instead of doing path*, we should traverse the children and wipe it.

Reply:
Done. Thank you.